src/Stratis.Bitcoin/P2P/PeerConnectorAddNode.cs

```diff
@@ -70,11 +76,7 @@ public override async Task OnConnectAsync()
                         PeerAddress peerAddress = this.peerAddressManager.FindPeer(ipEndpoint);
                         if (peerAddress != null && !this.IsPeerConnected(peerAddress.Endpoint))
                         {
-                            // Introduce a delay between attempts in case ConnectAsync fails instantly without the usual timeout.
```

Comment:
Ah Zeptin can we revert so that we return from if cancellation was requested? So that we eliminate the nested if.  Same with the other connector.